chat_exporter/construct/message.py

```diff
@@ -253,8 +253,12 @@ async def _gather_user_colour(self, author: discord.Member):
     async def _gather_user_icon(self, author: discord.Member):
         member = await self._gather_member(author)
 
-        if member and member.display_icon:
-            return f"<img class='chatlog__role-icon' src='{member.display_icon}' alt='Role Icon'>"
+        if member:
+            if hasattr(member, "display_icon") and member.display_icon:
+                return f"<img class='chatlog__role-icon' src='{member.display_icon}' alt='Role Icon'>"
+            elif hasattr(member, "top_role") and member.top_role:
```

Comment:
```
elif hasattr(member, "top_role") and member.top_role and member.top_role.icon:
```

Reply:
I haven't read much in to Pycord + Nextcord; but actually - couldn't you just go `elif(...) and member.top_role.icon`?